src/box/relay.cc

```diff
@@ -742,9 +769,16 @@ relay_reader_f(va_list ap)
 static inline void
 relay_send_heartbeat(struct relay *relay)
 {
+	if (relay->is_sending_tx)
+		return;
```

Comment:
Looks like this check as well as the added but never used `relay_next_vclock_sync` function belong to the next patch. This patch is supposed to just add syncs to heartbeats. I think we can do this unconditionally - a sync is tiny in comparison with a vclock, and heartbeats are sent relatively rarely.

Reply:
Done.